src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -156,8 +156,50 @@ public static Date getMsOffsetToCurrentTimeInUserTimeZone(int offset, double tim
     }
 
     public static Calendar convertToUserTimeZone(Calendar time, double timeZone) {
-        time.add(Calendar.MILLISECOND, (int) (60 * 60 * 1000 * timeZone));
-        return time; // for chaining
+        int serverOffset = time.getTimeZone().getRawOffset();
+        int sessionOffset = (int) (60 * 60 * 1000 * timeZone);
+        int offset = sessionOffset - serverOffset;
+        TimeZone sessionTimeZone = getTimeZone(timeZone);
```

Comment:
The problem with this solution is the effect is more far-reaching than you initially thought. If you do a workspace-wide search for this method (`TimeHelper.convertToUserTimeZone`), you will see that the usage goes much more than just what this issue is about; and they happen to be on areas that are not very well-tested so I am a bit wary if the change is done now.
Let's keep it simple for now? If you want to try implementing the semantically nicer solution, feel free to work on it, and also examine the effects on other parts of the application.